### PR TITLE
Refactor and strip down Aries transport abstraction and further code trimming

### PR DESCRIPTION
**Pull request by nain-F49FF806**

### CODE REVIEW

**Comment by nain-F49FF806** [comment]
Develops on commits from #1029 
Hence WIP, otherwise it's ready for review.

**Comment by Patrik-Stas** [comment]
Hi Naian, indeed as you sugggested, if you base this on https://github.com/hyperledger/aries-vcx/pull/1029 I can do review for this before 1029 is merged

File: agents/rust/mediator/tests/common/agent_and_transport_utils.rs [hunk 1]
```diff
@@ -68,15 +74,57 @@ pub async fn send_message_and_pop_response_message(
...
-    agent
-        .pack_and_send_didcomm(message_bytes, our_verkey, their_diddoc, aries_transport)
+    // Wrap message in encrypted envelope
+    let EncryptionEnvelope(packed_message) = agent
+        .pack_didcomm(message_bytes, our_verkey, their_diddoc)
         .await
-        .map_err(|err| GenericStringError { msg: err })?;
+        .map_err(|e| GenericStringError { msg: e.to_string() })?;
+    let packed_json = serde_json::from_slice(&packed_message)?;
+    info!(
```
Comment (Patrik-Stas) [inline agents/rust/mediator/tests/common/agent_and_transport_utils.rs:83]: Log as part of `send_aries_envelope`, but I'd suggest at least debug (or lower)

File: agents/rust/mediator/tests/common/agent_and_transport_utils.rs [hunk 2]
```diff
@@ -68,15 +74,57 @@ pub async fn send_message_and_pop_response_message(
...
+        agent_verkey,
+        mediator_diddoc,
+    )
+    .await
+    .unwrap();
+    // extract routing parameters
+    if let MediatorCoordMsgEnum::MediateGrant(grant_data) =
+        serde_json::from_str(&response_message).unwrap()
+    {
+        info!("Grant Data {:?}", grant_data);
```
Comment (Patrik-Stas) [inline agents/rust/mediator/tests/common/agent_and_transport_utils.rs:122]: the response is already logged in `send_aries_envelope`, which is called down the line calling `send_message_and_pop_response_message` on line `109`

**Review by Patrik-Stas — CHANGES_REQUESTED** [review]
image
The best kind of PRs  👍 

Left just a minor comments about logging

File: agents/rust/mediator/tests/common/agent_and_transport_utils.rs (hunk 1 above)
Comment (nain-F49FF806) [reply agents/rust/mediator/tests/common/agent_and_transport_utils.rs]: Done.

File: agents/rust/mediator/tests/common/agent_and_transport_utils.rs (hunk 2 above)
Comment (nain-F49FF806) [reply agents/rust/mediator/tests/common/agent_and_transport_utils.rs:118]: `send_aries_envelope` logs the encrypted response envelope. I've updated the log string to make it clear. 
Here we are logging the unencrypted and decoded message data.

**Review by Patrik-Stas — APPROVED** [review]

**Pull request merged**